### PR TITLE
Weather icon

### PR DESCRIPTION
**Pull request by ahujaradhika**
Calling OpenWeather API for current temperature and weather icons from their website

### CODE REVIEW

File: src/app/navbar/navbar.component.html [hunk 1]
```diff
@@ -16,6 +16,10 @@
     <div class="collapse navbar-collapse menu" id="navbarSupportedContent" [collapse]="isCollapsed">
         <ul class="navbar-nav mr-auto">
         </ul>
+        <span class="navbar-text">
+            <img src='http://openweathermap.org/img/w/{{icon}}.png' alt='Icon depicting current weather.'>
+        </span>
+        <span class="navbar-text">{{temperature}}°F</span>
```
Comment (TrinaKat) [inline src/app/navbar/navbar.component.html]: These can go in the same span

Comment (TrinaKat) [inline src/app/navbar/navbar.component.html]: Make it so when you click no the span, it switches between Fahrenheit and Celsius

File: src/app/navbar/navbar.component.ts [hunk 2]
```diff
@@ -12,9 +12,18 @@ export class NavbarComponent implements OnInit {
     @Output() changeView: EventEmitter<string> = new EventEmitter();
 
     constructor(private _eventService: EventService, private _categService: CategoryService) { }
-    ngOnInit() { }
-
+    temperature: string;
```
Comment (TrinaKat) [inline src/app/navbar/navbar.component.ts]: Make these explicitly public variables under the export statement

e.g.
export class NavbarComponent implements OnInit {
    public icon: string;

File: src/app/navbar/navbar.component.ts [hunk 3]
```diff
@@ -12,9 +12,18 @@ export class NavbarComponent implements OnInit {
...
+    icon: string;
```
Comment (TrinaKat) [inline src/app/navbar/navbar.component.ts]: Be more specific about variable purpose, e.g. "weatherIcon"

File: src/app/navbar/navbar.component.ts [hunk 4]
```diff
@@ -12,9 +12,18 @@ export class NavbarComponent implements OnInit {
...
     isCollapsed: boolean = true;
+    ngOnInit() { 
+      // our call back function 
+      var tempCallback = function callback(temperature, icon) : void{
```
Comment (TrinaKat) [inline src/app/navbar/navbar.component.ts]: Not sure we need to wrap this in a callback, can just call a function that sets the variables themselves (e.g. just make this.getTemperature() directly modify this.temperature and this.icon

File: src/app/navbar/navbar.component.ts [hunk 5]
```diff
@@ -47,4 +56,24 @@ export class NavbarComponent implements OnInit {
...
 
-}
+    getTemperature(callback): void{
+      var request = new XMLHttpRequest();
+      var data;
+      var temperature;
+      var icon;
+
+      request.open('GET', 'http://api.openweathermap.org/data/2.5/weather?units=imperial&zip=90024,us&APPID=bc6a73dfabbd4e6c9006a835d00589f2', true);
+      request.onreadystatechange = function(){
```
Comment (TrinaKat) [inline src/app/navbar/navbar.component.ts]: This is actually an overcomplicated way to make an api request, just use this.http.get. You may need to add that library, but otherwise the format is like other API requests in our code

File: src/app/navbar/navbar.component.ts [hunk 6]
```diff
@@ -47,4 +56,24 @@ export class NavbarComponent implements OnInit {
...
     }
 
-}
+    getTemperature(callback): void{
+      var request = new XMLHttpRequest();
+      var data;
+      var temperature;
+      var icon;
+
+      request.open('GET', 'http://api.openweathermap.org/data/2.5/weather?units=imperial&zip=90024,us&APPID=bc6a73dfabbd4e6c9006a835d00589f2', true);
```
Comment (TrinaKat) [inline src/app/navbar/navbar.component.ts]: Rather than have this really long request url, break it up into variables

e.g. API_KEY, baseWeatherUrl, zipcode
then you can form the query like "let weatherQuery = `${baseWeatherUrl}?units=imperial&zip=${zipcode},us&APPID=${apiKey}`;"

File: src/app/navbar/navbar.component.html [hunk 7]
```diff
@@ -16,6 +16,10 @@
     <div class="collapse navbar-collapse menu" id="navbarSupportedContent" [collapse]="isCollapsed">
         <ul class="navbar-nav mr-auto">
         </ul>
+        <span class="navbar-text">
+            <img src='http://openweathermap.org/img/w/{{icon}}.png' alt='Icon depicting current weather.'>
```
Comment (TrinaKat) [inline src/app/navbar/navbar.component.html]: We will want to fix this styling as well, it's making the navbar too tall/fat and has weird whitespace

Comment (TrinaKat) [reply src/app/navbar/navbar.component.html]: image
Also let's round the degree so it looks like a user would expect

File: src/app/navbar/navbar.component.ts [hunk 8]
```diff
@@ -1,6 +1,12 @@
 import { Component, OnInit, Output, EventEmitter } from '@angular/core';
 import { CategoryService } from '../category.service';
 import { EventService } from '../event.service';
+import { HttpClient } from '@angular/common/http';
+
+const API_KEY = "bc6a73dfabbd4e6c9006a835d00589f2";
```
Comment (TrinaKat) [inline src/app/navbar/navbar.component.ts]: These can be set in the actual function they're used, just as local vars.

File: src/app/navbar/navbar.component.ts [hunk 9]
```diff
@@ -47,4 +62,29 @@ export class NavbarComponent implements OnInit {
...
-}
+    getTemperature(): void { 
+      var weatherQuery = `${baseWeatherUrl}?units=${defaultTemperatureUnits}&zip=${zipcode},us&APPID=${API_KEY}`;
+
+      this.http.get(weatherQuery).subscribe(weatherData => {
+        //set default temperature to Fahrenheit
+        this.temperature = String(Math.round((weatherData['main']['temp']))) + "°F";
+        this.weatherIcon = weatherData['weather'][0]['icon'];
+
+        this.fahrenheit = Math.round(weatherData['main']['temp']);
```
Comment (TrinaKat) [inline src/app/navbar/navbar.component.ts]: Just to make this a bit more efficient just set:

let temp = weatherData...
this.fahrenheit = Math.round(temp);
this.celsius =  Math.round(temp - 32efwefwew);

this.temperature = String(this.fahrenheit) + *F
this.weatherIcon ...

File: src/app/navbar/navbar.component.ts [hunk 10]
```diff
@@ -47,4 +62,29 @@ export class NavbarComponent implements OnInit {
...
+        this.celsius = Math.round((weatherData['main']['temp'] - 32)/1.8);
+      });
+
+    }
+
+    isFahrenheit: boolean = true;
```
Comment (TrinaKat) [inline src/app/navbar/navbar.component.ts]: you can declare this up with the public vars. this can be private tho

File: src/app/navbar/navbar.component.ts (hunk 9 above)
Comment (TrinaKat) [reply src/app/navbar/navbar.component.ts]: actually instead of making fahrenheit/celsius ints, make them strings with *F and *C. so you dont even need to concatenate in future

File: src/app/navbar/navbar.component.ts [hunk 11]
```diff
@@ -47,4 +59,40 @@ export class NavbarComponent implements OnInit {
...
 
-}
+    getTemperature(): void { 
+
+      const API_KEY = "bc6a73dfabbd4e6c9006a835d00589f2";
+      const zipcode = "90024";
+      const baseWeatherUrl = "http://api.openweathermap.org/data/2.5/weather";
+      const defaultTemperatureUnits = "imperial";
+
+      var weatherQuery = `${baseWeatherUrl}?units=${defaultTemperatureUnits}&zip=${zipcode},us&APPID=${API_KEY}`;
```
Comment (tanzeelak) [inline src/app/navbar/navbar.component.ts]: use `let` instead of `var`. `var` is function scoped and `let` is block scoped. Basically, `var` ambiguously acts as a global variable. This isn't consistent in the code yet, so that's a future thing to improve later

File: src/app/navbar/navbar.component.ts [hunk 12]
```diff
@@ -47,4 +59,40 @@ export class NavbarComponent implements OnInit {
...
+        //set default temperature to Fahrenheit
+        if (this.isFahrenheit)
+          this.temperature = String(this.fahrenheit);
+        else
+          this.temperature = String(this.celsius);
+
+        this.weatherIcon = weatherData['weather'][0]['icon'];
+      });
+
+      console.log("updating temperature");
```
Comment (tanzeelak) [inline src/app/navbar/navbar.component.ts]: Delete console.log()'s when you submit for review. Logging slows down the program (but gr8 for debugging).

File: src/app/navbar/navbar.component.ts [hunk 13]
```diff
@@ -47,4 +59,40 @@ export class NavbarComponent implements OnInit {
...
+    }
+
+    switchTemperature(): void {
+
```
Comment (tanzeelak) [inline src/app/navbar/navbar.component.ts]: Del unnecessary whitespace (empty lines within functions)
